### DISCUSSION TITLE
Obtain filtered data in shared queries.

### DISCUSSION THREAD

**Discussion by SpaceOso** [TanStack/query#2219] [Q&A] [answered]
In one useQuery we are doing some heavy filtering. Using the select option to do this.

I now need that same selected data in another component. If I just use the same query id I get the raw data. I'm sure I'm doing something wrong, but how do you obtain only that selected data when you want to reuse the same query in a different component?

**Comment by TkDodo** [comment, marked as answer]
Write a custom hook that encapsulates useQuery with the select function and call it twice :)

**Reply by SpaceOso** [reply]
I run into the following error:

`  Line 112:10:  React Hook "useQuery" is called in function "getEnergyDeliveredQuery" which is neither a React function component or a custom React Hook function  react-hooks/rules-of-hooks
`
If I create a new file to share useQuery hooks :/

**Reply by SpaceOso** [reply]
Figured it out, had to rename my function to 

`UseGetEnergyDeliveredQuery`

Error I was seeing was a linting issue

It works now, thanks @TkDodo !!
